worlds/messenger/__init__.py

```diff
@@ -116,44 +221,132 @@ def create_items(self) -> None:
             total_seals = min(len(self.multiworld.get_unfilled_locations(self.player)) - len(itempool),
                               self.options.total_seals.value)
             if total_seals < self.total_seals:
-                logging.warning(f"Not enough locations for total seals setting "
-                                f"({self.options.total_seals}). Adjusting to {total_seals}")
+                logging.warning(
+                    f"Not enough locations for total seals setting "
+                    f"({self.options.total_seals}). Adjusting to {total_seals}"
+                )
                 self.total_seals = total_seals
             self.required_seals = int(self.options.percent_seals_required.value / 100 * self.total_seals)
 
             seals = [self.create_item("Power Seal") for _ in range(self.total_seals)]
-            for i in range(self.required_seals):
-                seals[i].classification = ItemClassification.progression_skip_balancing
             itempool += seals
 
+        self.multiworld.itempool += itempool
         remaining_fill = len(self.multiworld.get_unfilled_locations(self.player)) - len(itempool)
         if remaining_fill < 10:
             self._filler_items = self.random.choices(
-                                      list(FILLER)[2:],
-                                      weights=list(FILLER.values())[2:],
-                                      k=remaining_fill
+                list(FILLER)[2:],
+                weights=list(FILLER.values())[2:],
+                k=remaining_fill
             )
-        itempool += [self.create_filler() for _ in range(remaining_fill)]
+        filler = [self.create_filler() for _ in range(remaining_fill)]
 
-        self.multiworld.itempool += itempool
+        self.multiworld.itempool += filler
 
     def set_rules(self) -> None:
         logic = self.options.logic_level
         if logic == Logic.option_normal:
             MessengerRules(self).set_messenger_rules()
-        elif logic == Logic.option_hard:
-            MessengerHardRules(self).set_messenger_rules()
         else:
-            MessengerOOBRules(self).set_messenger_rules()
+            MessengerHardRules(self).set_messenger_rules()
+        # else:
+        #     MessengerOOBRules(self).set_messenger_rules()
+
+        add_closed_portal_reqs(self)
+        # i need portal shuffle to happen after rules exist so i can validate it
+        attempts = 5
```

Comment:
it's random fill and there's only 6 portals so it's really fast. even then it should only attempt more than once if the player uses plando, which is what the failsafe is for; if the player uses plando and creates a combo it can't solve.